storage: TestStoreRangeRemoveDead is flaky

https://circleci.com/gh/cockroachdb/cockroach/16885

Test is sitting on its WaitGroup:

```
goroutine 14692 [semacquire]:
sync.runtime_Semacquire(0xc82139f62c)
	/usr/local/go/src/runtime/sema.go:47 +0x26
sync.(*WaitGroup).Wait(0xc82139f620)
	/usr/local/go/src/sync/waitgroup.go:127 +0xb4
github.com/cockroachdb/cockroach/storage_test.TestStoreRangeRemoveDead(0xc821691ef0)
	/go/src/github.com/cockroachdb/cockroach/storage/client_raft_test.go:1168 +0x617
testing.tRunner(0xc821691ef0, 0x23ab710)
	/usr/local/go/src/testing/testing.go:473 +0x98
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:582 +0x892
```


```
I160426 14:50:28.310914 kv/txn_coord_sender.go:286  txn coordinator: 0.05 txn/sec, 100.00/0.00/0.00/0.00 %cmmt/cmmt1pc/abrt/abnd, 0/0/0 avg/σ/max duration, 0.0/0.0/0 avg/σ/max restarts (2 samples)
I160426 14:50:28.312368 kv/txn_coord_sender.go:286  txn coordinator: 0.00 txn/sec, 0.00/0.00/0.00/0.00 %cmmt/cmmt1pc/abrt/abnd, 0/0/0 avg/σ/max duration, 0.0/0.0/0 avg/σ/max restarts (0 samples)
I160426 14:50:28.327542 kv/txn_coord_sender.go:286  txn coordinator: 0.00 txn/sec, 0.00/0.00/0.00/0.00 %cmmt/cmmt1pc/abrt/abnd, 0/0/0 avg/σ/max duration, 0.0/0.0/0 avg/σ/max restarts (0 samples)
panic: test timed out after 1m10s

goroutine 14940 [running]:
panic(0x14e9560, 0xc820cae540)
	/usr/local/go/src/runtime/panic.go:464 +0x3e6
testing.startAlarm.func1()
	/usr/local/go/src/testing/testing.go:725 +0x14b
created by time.goFunc
	/usr/local/go/src/time/sleep.go:129 +0x3a
```

--- a/storage/client_raft_test.go
+++ b/storage/client_raft_test.go
@@ -1208,47 +1208,13 @@ func TestStoreRangeRemoveDead(t *testing.T) {
 	mtc.Start(t, 3)
 	defer mtc.Stop()
 
-	var sgs []*gossiputil.StoreGossiper
-	for _, g := range mtc.gossips {
-		sgs = append(sgs, gossiputil.NewStoreGossiper(g))
-	}
-
 	// Replicate the range to all stores.
 	replica := mtc.stores[0].LookupReplica(roachpb.RKeyMin, nil)
 	mtc.replicateRange(replica.RangeID, 1, 2)
 
-	// Initialize the gossip network.
-	var storeIDs []roachpb.StoreID
-	for _, s := range mtc.stores {
-		storeIDs = append(storeIDs, s.StoreID())
-	}
-
-	// Gossip all stores and wait for callbacks to be run. This is
-	// tricky since we have multiple gossip objects communicating
-	// asynchronously. We use StoreGossiper to track callbacks but we
-	// need to set up all the callback tracking before any stores are
-	// gossiped.
-	var readyWG sync.WaitGroup
-	var doneWG sync.WaitGroup
-	readyWG.Add(len(sgs))
-	doneWG.Add(len(sgs))
-	for _, sg := range sgs {
-		go func(sg *gossiputil.StoreGossiper) {
-			ready := false
-			sg.GossipWithFunction(storeIDs, func() {
-				if !ready {
-					readyWG.Done()
-					ready = true
-				}
-			})
-			doneWG.Done()
-		}(sg)
-	}
-	readyWG.Wait()
 	for _, s := range mtc.stores {
 		s.GossipStore()
 	}
-	doneWG.Wait()
 
 	rangeDesc := getRangeMetadata(roachpb.RKeyMin, mtc, t)
 	if e, a := 3, len(rangeDesc.Replicas); e != a {

